src/App.tsx

```diff
@@ -21,65 +21,76 @@ import MushroomGame from "./pages/MushroomGame";
 
 //protected route for admin routes
 const AdminProtectedRoute = ({ children }: { children: React.ReactNode }) => {
-  const { user } = useAuth(); 
+ const { user } = useAuth();
 
-  if (!user) {  //redirect to login if user not authenticated
-    return <Navigate to="/login" />;
-  }
-  return <>{children}</>; //allow access if authenticated
+ if (!user) {
+  //redirect to login if user not authenticated
+  return <Navigate to="/login" />;
+ }
+ return <>{children}</>; //allow access if authenticated
 };
 
 const App = () => {
-  let LanguageContext = createContext("");
-  const [currentLang, setCurrentLang] = useState("");
+ let LanguageContext = createContext("");
+ const [currentLang, setCurrentLang] = useState("");
 
-  // Get the language from local storage so it doesn't change on refresh
-  useEffect(() => {
-    const lang = sessionStorage.getItem("lang");
-    if (lang) {
-      setCurrentLang(lang); 
-      LanguageContext = createContext(lang);
-    } else {
-      setCurrentLang("jp");
-      LanguageContext = createContext("jp");
-    }
-  }, []);
+ // Get the language from local storage so it doesn't change on refresh
+ useEffect(() => {
+  const lang = sessionStorage.getItem("lang");
+  if (lang) {
+   setCurrentLang(lang);
+   LanguageContext = createContext(lang);
+  } else {
+   setCurrentLang("jp");
+   LanguageContext = createContext("jp");
+  }
+ }, []);
```

Comment:
The comment says local storage but you are using session storage. Actually I think in your use case local storage would be great. Usually the user of the computer would want it to be the same always for the site. And not have to set always on tab visit. And it isn't private data. Just a locale key so it is safe to put in local storage.

Reply:
I changed it.